Source/WebCore/page/DragController.cpp

```diff
@@ -445,7 +447,7 @@ DragHandlingMethod DragController::tryDocumentDrag(LocalFrame& frame, const Drag
         }
 
         IntPoint point = frameView->windowToContents(dragData.clientPosition());
-        Element* element = elementUnderMouse(m_documentUnderMouse.get(), point);
+        RefPtr element = elementUnderMouse(*m_documentUnderMouse, point);
```

Comment:
This is still not safe, you need a RefPtr to the documentUnderMouse on the stack. So what I would do is:
1. Add a `RefPtr<Document> protectedDocumentUnderMouse() const` getter to the class
2. Call it here:  `RefPtr element = elementUnderMouse(*protectedDocumentUnderMouse(), point);`

Reply:
I have done as you suggested, but why not call protectedDocumentUnderMouse() at the top of this function and use the result instead of m_documentUnderMouse throughout?

Reply:
If `m_documentUnderMouse` is used several times in the function and you're sure its value doesn't change, then storing it in a local is indeed better. You wouldn't call `protectedDocumentUnderMouse()` though. The `protectedFoo()` functions are there to *avoid* local variables. You would use:
```
RefPtr documentUnderMouse = m_documentUnderMouse;
```

Reply:
It does seem like it can change at least once so I'll leave that refactoring for someone more familiar with our drag & drop code. Thanks!